### PR TITLE
Add support for alphanumeric map seed

### LINKED ISSUE #36: Support Alphanumeric seeds
### Discussed in https://github.com/jacobsjo/mc-datapack-map/discussions/35

<div type='discussions-op-text'>

<sup>Originally posted by **veradalelabs** February 14, 2024</sup>
Is it possible to use an alphanumeric value for the world seed?

I can type something in, but it just reverts to zero.

Not sure if this is an issue, or working as intended.</div>

### PR DESCRIPTION
**Pull request by CrazyChaoz**
This SHOULD implement/fix #36

### CODE REVIEW

**Comment by CrazyChaoz** [comment]
Currently it immediately converts a string into the real seed, but plugging the function into a later stage would add more visual clarity.

**Comment by jacobsjo** [comment]
Looks good! I didn't really like the way you did the check if it is a number, since we are dealing with bigints here (though I guess it did work). I already had that check with the try catch below, so I just moved all that up into the function. 

> Currently it immediately converts a string into the real seed, but plugging the function into a later stage would add more visual clarity.

I think that is fine. The game does the same thing when you run `/seed`.

**Comment by CrazyChaoz** [comment]
> I think that is fine. The game does the same thing when you run /seed.

Then a small onHover tooltip, like with a html ```title``` tag would probably help users 
Something like ```alphanumeric seeds get automatically converted to real seeds```
> I already had that check with the try catch below, so I just moved all that up into the function.

Fine by me, it was just the first concise check in js I found

**Comment by jacobsjo** [comment]
One problem still remains: When entering values in other bases (such as `0xFF`), the `BigInt()` converts it into a number (i.e. 256), but Minecraft calculates the hash.

**Comment by CrazyChaoz** [comment]
I just checked - my initial number detection also recognizes it as a number ....
Are you fine with a regex? 
I assume Minecraft just converts any string with a non \d digit.
```    
if (/^\d+$/.test(input_string)) {
    return input_string
}
```

File: src/components/SettingsPanel.vue [hunk 1]
```diff
@@ -19,6 +19,22 @@ datapackStore.$subscribe(async (mutation, state) => {
 function randomizeSeed() {
     settingsStore.seed = random.nextLong()
 }
+
+function parseSeed(input: string): bigint {
+    if (/^\d+$/.test(input)) {
```
Comment (jacobsjo) [inline src/components/SettingsPanel.vue:24]: This needs to accept a leading `-` and `+`.

Comment (CrazyChaoz) [reply src/components/SettingsPanel.vue:24]: AAAAAAAHHHHH

Comment (CrazyChaoz) [reply src/components/SettingsPanel.vue]: could you quickly check what happens with +/- in non-digit seeds? do i have to add it again ?

Comment (jacobsjo) [reply src/components/SettingsPanel.vue]: I non-digit seeds it just treated as part of the string to get the hash of, so the current behavior is correct. :+1:

**Comment by jacobsjo** [comment]
Just added one more check, since the game doesn't allow numbers outside the -2^63 to 2^63-1 range, and just treats them as strings instead.

**Comment by jacobsjo** [comment]
Thanks for the PR!

**Pull request merged**